src/kernels/ipywidgets-message-coordination/notebookIPyWidgetCoordinator.ts

```diff
@@ -104,14 +105,16 @@ export class NotebookIPyWidgetCoordinator {
     private readonly notebookEditors = new WeakMap<NotebookDocument, NotebookEditor[]>();
     constructor(
         @inject(IServiceContainer) private readonly serviceContainer: IServiceContainer,
-        @inject(IDisposableRegistry) disposableRegistry: IDisposableRegistry,
+        @inject(IDisposableRegistry) private readonly disposableRegistry: IDisposableRegistry,
         @inject(IAsyncDisposableRegistry) private readonly asyncDisposableRegistry: IAsyncDisposableRegistry,
         @inject(IVSCodeNotebook) private readonly notebook: IVSCodeNotebook,
         @inject(INotebookControllerManager) private readonly controllerManager: INotebookControllerManager
     ) {
-        notebook.onDidChangeVisibleNotebookEditors(this.onDidChangeVisibleNotebookEditors, this, disposableRegistry);
-        notebook.onDidCloseNotebookDocument(this.onDidCloseNotebookDocument, this, disposableRegistry);
-        controllerManager.onNotebookControllerSelected(this.onDidSelectController, this, disposableRegistry);
```

Comment:
Code was in the ctor and when we got this instance from test code, this class would get instantiated and things work.
Moved into an activate function to be more specific instead of relying on side effects of ctor